hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactInternalGenericRecord.java

```diff
@@ -420,6 +420,38 @@ public <T> T getObject(@Nonnull String fieldName) {
         return (T) getVariableSize(fieldName, COMPACT, in -> serializer.read(in, schemaIncludedInBinary));
     }
 
+    /**
+     * {@link CompactInternalGenericRecord#getObject(String)} with known field descriptor. Introduced not to call
+     * {@link com.hazelcast.internal.serialization.impl.compact.Schema#getField(String)} twice if the field
+     * descriptor is known.
+     *
+     * @param fd Field descriptor of the field. This field descriptor have to be retrieved via
+     * {@link CompactInternalGenericRecord#getField(String)} method of this {@link CompactInternalGenericRecord}.
+     * @return a nested field as a concrete deserialized object rather than generic record
+     * @throws HazelcastSerializationException if the field name does not exist in the class definition/schema or
+     *                                         the type of the field does not match the one in the class definition/schema.
+     */
+    @Nullable
+    public <T> T getObject(@Nonnull FieldDescriptor fd) {
+        return (T) getVariableSize(fd, in -> serializer.read(in, schemaIncludedInBinary));
```

Comment:
does ıt only work for var sized? what about fixed sized?

Reply:
We are calling `readAsLeafObjectOnQuery` in `GenericRecordQueryReader`. It is defined with default implementation for fixed fields [here](https://github.com/hazelcast/hazelcast/blob/4690a0ff0ca99d290b5649f4f91f47b3172bf6b7/hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/FieldKindBasedOperations.java#L35). `getObject` is only called for compact field kind. 

Does this answers your question?